snorkel/classification/snorkel_classifier.py

```diff
@@ -328,10 +322,23 @@ def predict(
         prob_dict_list: Dict[str, List[torch.Tensor]] = defaultdict(list)
 
         for batch_num, (X_batch_dict, Y_batch_dict) in enumerate(dataloader):
-            prob_batch_dict = self._calculate_probs(X_batch_dict, Y_batch_dict.keys())
-            for task_name, Y in Y_batch_dict.items():
-                prob_dict_list[task_name].extend(prob_batch_dict[task_name])
-                gold_dict_list[task_name].extend(Y.cpu().numpy())
+            labels_to_tasks = self._get_labels_to_tasks(Y_batch_dict.keys())
+            prob_batch_dict = self._calculate_probs(
+                X_batch_dict, labels_to_tasks.values()
+            )
+            for label_name, Y in Y_batch_dict.items():
+                task_name = labels_to_tasks[label_name]
+
+                # Filter out inactive probs and golds
+                active = get_active_mask(Y)
+
+                # Gather the output from the appropriate task
+                probs = prob_batch_dict[task_name][active].cpu().numpy()
```

Comment:
The issue I see here (and I believe the reason why active filtering in this step was removed) is that the user loses track of which examples correspond to which predictions/golds. If I have metadata corresponding to my 50 data points, then I run predict and get back 30 preds/golds because 20 were filtered, how do I map them to the rest of my data? One option is to tell the user to pass through UIDs in their X_dict. But alternatively, could we have `predict()` just return ABSTAIN  labels for those examples and then deal with not including those in our metrics using the filter_dict in metric_score?

Reply:
Just seems like if I tell a method to make predictions for 50 datapoints, I should get back 50 predictions, even if some of those are "I don't know".

Reply:
what setting has this been a problem in?
in general though, happy to make the change to support filtering in the scorer but surface abstain in predictions as well

Reply:
Jared today was saying that he won't use a system that doesn't have support UIDs, since he's been burned too many times by hidden shuffles that happen and screw up alignment between data and labels. We don't have that here, per se, but it's a similar issue, I think. It would be a confusing error if a person wanted to do something with their predictions and somewhere in the middle of their pipeline their dataset size shrunk.

Reply:
Okay, yes—it will probably also be easier to implement UIDs (potentially down the line) if this happens in the Scorer.